moveit_core/collision_detection/include/moveit/collision_detection/test_collision_common_panda.h

```diff
@@ -236,6 +236,43 @@ TYPED_TEST_P(CollisionDetectorPandaTest, DistanceSelf)
   EXPECT_NEAR(res.distance, 0.022, 0.001);
 }
 
+/** \brief Tests if nearest points are computed correctly. */
+TYPED_TEST_P(CollisionDetectorPandaTest, DistancePoints)
+{
+  // Adding the box right in front of the robot hand
+  shapes::Shape* shape = new shapes::Box(0.1, 0.1, 0.1);
+  shapes::ShapeConstPtr shape_ptr(shape);
+
+  Eigen::Isometry3d pos{ Eigen::Isometry3d::Identity() };
+  pos.translation().x() = 0.43;
+  pos.translation().y() = 0;
+  pos.translation().z() = 0.55;
+  this->cenv_->getWorld()->addToObject("box", shape_ptr, pos);
+
+  collision_detection::DistanceRequest req;
+  req.acm = &*this->acm_;
+  req.type = collision_detection::DistanceRequestTypes::SINGLE;
+  req.enable_nearest_points = true;
+  req.max_contacts_per_body = 1;
+
+  collision_detection::DistanceResult res;
+  this->cenv_->distanceRobot(req, res, *this->robot_state_);
+
+  // Check that all points reported on "box" are actually on the box
```

Comment:
If I understand correctly, the other option is that the reported points are on the hand? If that's right, please add comment:

`// The other option is that reported points are on the hand`

Reply:
I added `... and not on the robot` which should make it more clear.